include/cutlass/detail/helper_macros.hpp

```diff
@@ -146,7 +146,13 @@ namespace cutlass {
 
 ////////////////////////////////////////////////////////////////////////////////////////////////////
 
-#if (201700L <= __cplusplus)
+#ifdef _MSVC_LANG  
+#define MSVC_LANG _MSVC_LANG
+#else
+#define MSVC_LANG 0
+#endif
+
+#if (201700L <= __cplusplus || 201700L <= MSVC_LANG)
```

Comment:
1. The C++ Standard reserves identifiers with `_` prefix for the implementation's use.
2. This change defines a macro with a short name and no library prefix (e.g.,` CUTLASS_`) for all downstream code.
3. It looks like this macro is only used on line 155.

As a result, should we consider not defining `MSVC_LANG`, and instead expressing the idea inline?

```suggestion
#if (defined(_MSVC_LANG) && (201700L <= MSVC_LANG)) || (201700L <= __cplusplus)
```

Reply:
Alternately, should we consider name-prefixing the macro and defining it as "the way to test the language version for any compiler"?

```suggestion
#if defined(_MSVC_LANG)
#  define CUTLASS_CPLUSPLUS _MSVC_LANG
#else
#  define CUTLASS_CPLUSPLUS __cplusplus
#endif

#if (201700L <= CUTLASS_CPLUSPLUS)
```